Add uv package mananger and restructure the folders in mcp-clients

santhoshsai85 opened the pull request:
- Adding uv packager
- Restructure the from mcp_clients to mcp-clients
- Update the Dockerfiles and READMEs

This PR partially addresses #6 includes only mcp_clients

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Klavis-AI/klavis?pullRequest=13) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Klavis-AI/klavis?pullRequest=13) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Klavis-AI/klavis?pullRequest=13) it.</sub>

zihaolin96 commented:
I think using underscore `mcp_clients` is best practice, actually we previous got some module import issue with `mcp-clients`

zihaolin96 commented on mcp-clients/pyproject.toml:
```diff
@@ -0,0 +1,29 @@
+[project]
```
Thanks! was trying to change from requirements.txt to poetry

santhoshsai85 commented:
> I think using underscore `mcp_clients` is best practice, actually we previous got some module import issue with `mcp-clients`

Hey @zihaolin96, `mcp-clients` is just the project name(preferred standard) andthe package is still named as `mcp_clients` which is under `mcp-clients/src` directory. All the files are under `mcp_clients` so you would not be facing module import issues.

zihaolin96 commented:
> > I think using underscore `mcp_clients` is best practice, actually we previous got some module import issue with `mcp-clients`
> 
> Hey @zihaolin96, `mcp-clients` is just the project name(preferred standard) and the package is still named as `mcp_clients` which is under `mcp-clients/src` directory. All the files are under `mcp_clients` so you would not be facing module import issues.

Thanks for sharing the knowledge! 
not `uv` expert, but the current file structure looks a bit weird to me, e.g. 
```
mcp-clients/src/mcp_clients/discord_bot.py
``` 
any way we could improve this?

santhoshsai85 commented:
> > > I think using underscore `mcp_clients` is best practice, actually we previous got some module import issue with `mcp-clients`
> > 
> > 
> > Hey @zihaolin96, `mcp-clients` is just the project name(preferred standard) and the package is still named as `mcp_clients` which is under `mcp-clients/src` directory. All the files are under `mcp_clients` so you would not be facing module import issues.
> 
> Thanks for sharing the knowledge! not `uv` expert, but the current file structure looks a bit weird to me, e.g.
> 
> ```
> mcp-clients/src/mcp_clients/discord_bot.py
> ```
> 
> any way we could improve this?

This is the preferred industry standard for python project structuring, mainly its either `project-name/src/project_name` or `project-name/project_name`. It would help in easier maintenance, currently there are no tests, but generally tests would go under `mcp-clients/tests`. Similarly for any no src code related stuff.

xiangkaiz commented:
Just a random thought: What you do think of `klavis/src/mcp_clients/...` and `klavis/src/mcp_servers/...` instead of `mcp-clients/src/mcp_clients/`?

santhoshsai85 commented:
> Just a random thought: What you do think of `klavis/src/mcp_clients/...` and `klavis/src/mcp_servers/...` instead of `mcp-clients/src/mcp_clients/`?

@xiangkaiz That would make `klavis` as a project and it would contain a single `pyproject.toml` file which should include all the dependencies used in the packages under `src` folder(like `mcp_clients`, `mcp_servers`, etc), which is not ideal in this case. And the `mcp_servers` also contains a `go` language for github mcp server. That would not make any sense if we keep `klavis` as root project.

xiangkaiz commented:
@santhoshsai85 Make sense, thanks! I am away from keyboard now. I should be able to take a closer look at this PR by end of day today. Thanks!

santhoshsai85 commented:
> @santhoshsai85 Make sense, thanks! I am away from keyboard now. I should be able to take a closer look at this PR by end of day today. Thanks!

Sure, let me know if you are having any issue with setting up the project. It should work fine if you went ahead with updated readme's.

xiangkaiz commented:
Hi @santhoshsai85 ! Thank you for your patience and your work! I have 3 general comments:

1. I tried both docker and directly uv run and got the same error:
```
  File "/app/.venv/bin/discord_bot", line 4, in <module>
    from mcp_clients.discord_bot import main
  File "/app/src/mcp_clients/discord_bot.py", line 13, in <module>
    from mcp_clients.base_bot import BaseBot, BotContext
  File "/app/src/mcp_clients/base_bot.py", line 11, in <module>
    from mcp_clients.llms import Conversation, ChatMessage
  File "/app/src/mcp_clients/llms/__init__.py", line 1, in <module>
    from .anthropic import Anthropic
  File "/app/src/mcp_clients/llms/anthropic.py", line 8, in <module>
    from mcp_clients.llms import (
ImportError: cannot import name 'BaseLLM' from partially initialized module 'mcp_clients.llms' (most likely due to a circular import) (/app/src/mcp_clients/llms/__init__.py)
```

2. I thought about it and think the project and top level folder name could be `klavis-mcp-clients` instead of `mcp-clients`. Let me know what you think, thanks!
3. Could you update the readme for slack as well? You can leave the readme for web client because we are still testing it.

Thank you and let me know if you have any questions!

zihaolin96 commented:
+1 have the same error with uv run

santhoshsai85 commented:
1. I have pushed the changes to resolve the circular import issue. Let me know if you able to run them now.
2. Sure, I can rename it, but as its inside of `klavis` already doesn't it mean its klavis's mcp-clients. Let me know if you prefer  naming it as `klavis-mcp-clients`, I wil rename it.
3. I have updated readme for slack on initial github setup, not too sure on the docker run commands.

xiangkaiz commented on mcp-clients/src/mcp_clients/llms/__init__.py:
I am curious why we need to remove the content of this file.

santhoshsai85 replied:
Hey @xiangkaiz, the issue was `llms/__init__.py` importing the `anthropic` and `anthropic.py` was in-turn importing `BaseLLM` from `llms/__init__.py`. This caused circular dependency issue.

xiangkaiz reviewed (commented):
Thank you for the changes.

xiangkaiz commented:
For 2: I agree with you. No need to change.

For 3. @zihaolin96 could you take a look and test slack? I tested discord and it is working now.

zihaolin96 commented on mcp-clients/README-Slack.md:
```diff
@@ -5,10 +5,44 @@ This guide will help you set up your own Slack app and connect it to our applica
+- uv 0.6.14+ installed on your machine
 - Git repository cloned locally
 - ngrok installed (for local development)
 
```
nit: can you also add below discord env setup to slack as well?
```
### Method 2: Python Virtual Environment
1.  **Create and Activate Virtual Environment:**
    ```bash
    # Make sure to navigate to the root directory of the project (skip if already done)
    cd klavis/mcp-clients
    ```
    ```bash
    # Create environment (only needs to be done once)
    uv venv
    # Activate environment
    # Windows (Command Prompt/PowerShell):
    .venv\Scripts\activate
    # macOS/Linux (bash/zsh):
    source .venv/bin/activate
    ```
2.  **Install Dependencies:**
    ```bash
    uv sync
    ```
3.  **Run the Bot:**
    Ensure your `.env` file exists in the `klavis/mcp-clients` root and `src/mcp_clients/local_mcp_servers.json` is configured.
    ```bash
    uv run discord_bot

```

zihaolin96 commented on mcp-clients/src/mcp_clients/slack/oauth_routes.py:
```diff
@@ -1,23 +1,25 @@
 
 if USE_PRODUCTION_DB:
-    from database.database import create_mcp_client, insert_slack_client_auth
+    from mcp_clients.database import create_mcp_client, insert_slack_client_auth
```
nit:
```suggestion
    from mcp_clients.database.database import create_mcp_client, insert_slack_client_auth
```

zihaolin96 commented on mcp-clients/src/mcp_clients/slack/message_handlers.py:
```diff
@@ -1,25 +1,27 @@
 
 if USE_PRODUCTION_DB:
-    from database.database import get_mcp_client_id_by_slack_info
+    from mcp_clients.database import get_mcp_client_id_by_slack_info
```
```suggestion
    from mcp_clients.database.database import get_mcp_client_id_by_slack_info
```

zihaolin96 reviewed (approved):
Feel free to merge after resolve left comments! Thanks for amazing work!

santhoshsai85 commented:
> Feel free to merge after resolve left comments! Thanks for amazing work!

@zihaolin96 I don't see an option to merge the PR, as Im not added as contributor to the project. I have addressed the comments, so you are good the merge the PR.

zihaolin96 commented:
@santhoshsai85 Thanks! merged!

VAIBHAVSING commented:
@zihaolin96 please close this pr and it's referance issue
